src/components/App/index.tsx

```diff
@@ -107,7 +107,7 @@ async function compressImage(
 
   return new File(
     [compressedData],
-    sourceFilename.replace(/\..+$/, '.' + encoder.extension),
+    sourceFilename.replace(/.([^.]*)$/, `.${encoder.extension}`),
```

Comment:
Since we don't need to capture the group here, let's avoid capturing: `/.(?:[^.]*)$/`

In fact, we don't need the group at all: `/.[^.]*$/`

Reply:
Good catch @mathiasbynens I am sending in a fix now